app/controllers/unwithdraw_controller.rb

```diff
@@ -22,10 +22,16 @@ def unwithdraw
 
 private
 
-  def check_permission
-    return if current_user.has_permission?(User::MANAGING_EDITOR_PERMISSION)
-
+  def check_permissions
     @edition = Edition.find_current(document: params[:document])
-    render :non_managing_editor, status: :forbidden
+
+    if !current_user.has_permission?(User::MANAGING_EDITOR_PERMISSION) && !@edition.history_mode?
+      render :non_managing_editor, status: :forbidden
+      return
+    end
+
+    if !current_user.has_permission?(User::MANAGE_LIVE_HISTORY_MODE) && @edition.history_mode?
+      render "missing_permissions/update_history_mode", status: :forbidden
+    end
```

Comment:
I feel like this would be simpler if we did:

```
    if !current_user.has_permission?(User::MANAGE_LIVE_HISTORY_MODE) && @edition.history_mode?
      render "missing_permissions/update_history_mode", status: :forbidden
      return
    end

    if !current_user.has_permission?(User::MANAGING_EDITOR_PERMISSION)
      render :non_managing_editor, status: :forbidden
    end
```

Reply:
Ahhhhh, you're correct!